pylint/lint/pylinter.py

```diff
@@ -480,6 +480,9 @@ def register_checker(self, checker: checkers.BaseChecker) -> None:
             self.register_report(r_id, r_title, r_cb, checker)
         if hasattr(checker, "msgs"):
             self.msgs_store.register_messages_from_checker(checker)
+            for message in checker.messages:
+                if message.default_disabled:
+                    self.disable(message.msgid)
```

Comment:
It feel like it's the responsibility of the Message Store to not register disabled messages in ``register_messages_from_checker``.

Reply:
We want to register all messages. Otherwise the user isn't able to enable it via the config setting. It's the `MessageStateHandler` which deals with `enable`, `disable`, and `disable-next`.

Reply:
> Hmm, right. It feels wrong that this code is in `PyLinter`,

I don't think so. A few lines below we also disable messages if the complete checker is disabled.
https://github.com/PyCQA/pylint/blob/77e8ae4f3902697babb186482f577a2c8b1a3725/pylint/lint/pylinter.py#L483-L485

> ... it should probably happen somewhere in `pylint.message`. Do you think moving that to MessageStateHandler is possible ?

Not without some bigger refactoring. `MessageStateHandler` is part of the `pylint.lint` package and not related to `pylint.messages`. That's why `PyLinter` is the best place IMO. It's the connection between both.

Reply:
> PyLinter is the best place IMO. It's the connection between both.

It's the connection between everything and is still way too huge even after Daniel most welcome refactors to separate the configuration parsing. But yeah, if it's a huge refactor then let's not do it right now.